packages/components/src/popover/index.tsx

```diff
@@ -448,24 +491,28 @@ const Popover = (
 						`is-${ computedPlacement.split( '-' )[ 0 ] }`,
 					].join( ' ' ) }
 					style={ {
-						left: Number.isFinite( arrowData?.x )
-							? `${
-									arrowData.x +
-									( frameOffsetRef.current?.x ?? 0 )
-							  }px`
-							: '',
-						top: Number.isFinite( arrowData?.y )
-							? `${
-									arrowData.y +
-									( frameOffsetRef.current?.y ?? 0 )
-							  }px`
-							: '',
+						left:
+							typeof arrowData?.x !== 'undefined' &&
+							Number.isFinite( arrowData.x )
+								? `${
+										arrowData.x +
+										( frameOffsetRef.current?.x ?? 0 )
+								  }px`
+								: '',
+						top:
+							typeof arrowData?.y !== 'undefined' &&
+							Number.isFinite( arrowData.y )
+								? `${
+										arrowData.y +
+										( frameOffsetRef.current?.y ?? 0 )
+								  }px`
+								: '',
```

Comment:
Runtime change: somehow `Number.isFinite( arrowData.x )` was not guarding against `arrowData.x` not being `undefined`, and so I added an explicit check (same for `arrowData.y`)